x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/DataFrameAnalysis.java

```diff
@@ -83,6 +83,12 @@
      */
     boolean supportsInference();
 
+    /**
+     * @return the percentage of data to use for training
+     */
+    default double getTrainingPercent() {
```

Comment:
[nit] IMO it would be better to remove the `default` implementation from this interface and put it in `OutlierDetection`.